pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -104,18 +104,15 @@ public void init(HelixManager helixManager, TableConfig tableConfig) {
   @Override
   public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap) {
-    InstancePartitions instancePartitions = instancePartitionsMap.get(InstancePartitionsType.CONSUMING);
-    Preconditions.checkState(instancePartitions != null, "Failed to find CONSUMING instance partitions for table: %s",
-        _realtimeTableName);
+    Preconditions.checkState(instancePartitionsMap.size() == 1, "One instance partition type should be provided");
+    InstancePartitions instancePartitions = instancePartitionsMap.entrySet().iterator().next().getValue();
```

Comment:
We should check the `InstancePartitionsType` of the entry to decide how to assign the segment. The current logic is for `CONSUMING` type only, and we should add one for `COMPLETED` type, which should be similar to `OfflineSegmentAssignment.assignSegment()`